### DISCUSSION TITLE
Few ideas for the future improvement

### DISCUSSION THREAD

**Discussion by acc4github** [ZL-Audio/ZLEqualizer#61] [Ideas]
These are just ideas(that I'd personally want to have in ZL if possible), not a demand. Feel free to find it useful or not. :D 

■ General
- Preset saving feature, with an ability to set a default preset (for instance: HQ and SCG is on by default when a new ZL instance is loaded)
- Linear Phase option
- Mid/Side processing

■ UI Control
- Double click to disable/enable the band (bypass/on)
- Dynamic Link button for the compression band can be moved aside a bit and also be a bit smaller, so that it won't overlap with the bands when tweaking it around
- Ability for the small panel that displays the current band info to be grabbed&moved around, so that it won't overlap with the dynamic bands when tweaking it around

■ Dynamic Related
- Ability to see the Threshold for the compression in the analyzer
- Ability to link the target gain's q to the main gain's q as well (for the same reason for [this](https://github.com/ZL-Audio/ZLEqualizer/discussions/52)) *I think making the Dynamic Link more like checkbox-formed to select which feature you want to link to the main band would be the ultimate ideal
- Ability to Solo the frequency with the compression applied (The current behavior seems to just solo the frequency like the main band, without being able to listen to how the sound is processed)
- Delta Listen

**Comment by zsliu98** [comment]
Thanks for the suggestions.

General:

- Preset saving feature: Logic is simple, but GUI is too complex. I would leave that feature to DAWs (Reaper supports this).
- Linear Phase: cannot find a good algorithm to do linear phase dynamic :sweat:
- Mid/Side: you can select it from the bottom left panel

UI:

- double click to bypass/on: that might be a little counter-intuitive
- dynamic link button: I want to separate it from the dragger and make it easy to control ... let me think about it carefully
- small panel: you may click somewhere else to hide that panel. After that, you can adjust the dynamic band.

Dynamic related:

- threshold: that value does not make sense on the analyzer. It is the threshold of the short-term RMS of the band-pass filter, which is completely different from the result of FFT.
- link the target gain's q: when you adjust the knob with the mouse wheel, those two are linked.
- solo the frequency with the compression applied: AFAIK solo filters in most EQs are on the unprocessed main signal.
- delta listen: for any high-order filter (> 12 dB/oct, especially high/low pass), the delta signal is highly distorted. And I don't believe delta signal is helpful ...

**Reply by acc4github** [reply]
Just a few clarifications on the ideas:

---
> threshold: that value does not make sense on the analyzer. It is the threshold of the short-term RMS of the band-pass filter, which is completely different from the result of FFT.

I meant just like a pale line or something overlapped onto the analyzer area based on the threshold number.
![FL64_swLfexRy5B](https://github.com/ZL-Audio/ZLEqualizer/assets/77421012/f9496081-8016-4a5d-90e8-e89c33c61ecb)

---
> link the target gain's q: when you adjust the knob with the mouse wheel, those two are linked.

I meant the target eq's q, which is controlled by the yellow circle on the knob. I don't think it's linked to the main band's q. 
 
---
> solo the frequency with the compression applied: AFAIK solo filters in most EQs are on the unprocessed main signal.

I meant the dynamic band's solo filter, that is seperated from the main eq's solo filter. I think it'd be useful to hear how much dynamic is applied.
![FL64_oAgjqJxBRh](https://github.com/ZL-Audio/ZLEqualizer/assets/77421012/11296418-2c92-4303-86fe-903e605d8931)

---
All of these is probably kinda too niche of a need tho. I'm not demanding an answer, so feel free to just leave it on read. :>

**Reply by zsliu98** [reply]
- Yes, I understand. But that value should not be displayed on the spectrum. Even if the spectrum is below that line, the dynamic may still get triggered.
- If you adjust it with the mouse-wheel on the bottom gain/Q knob, you should adjust both the main & target at the same time. (But those two are not *linked*).
- Dynamic solo filter is on the side chain. You will hear the signal which triggers the dynamic, which will be completely different from the main-chain if you use an external side-chain.

**Reply by zsliu98** [reply]
An update on this: Since different Q values for main & target are less common, I have adjusted the spectrum mouse wheel control as follows:
- if you place the mouse cursor right above the selected band main/target/side dragger, mouse-wheel will adjust the Q of the main/target/side filter
- otherwise, mouse-wheel will adjust both main & target Q
Hopefully, it will make the Q adjustment of dynamic bands easier.
